lib/index.ts

```diff
@@ -23,15 +23,21 @@ export let schema: PrismaSchema | undefined;
  * Create a Prisma schema object.
  * @param props.datasource The datasource object information for Prisma.
  * @param props.generator The generator object information for Prisma.
+ * @param props.basePath The base path for the schema object, optional if you want a purely programmatically generated schema.
  * @returns The generated schema object.
  */
 export const createSchema = ({
-  basePath,
   datasource,
   generator,
+  basePath,
 }: CreateSchemaOptions) => {
-  schema = new PrismaSchema(basePath, datasource, generator);
+  if (!basePath) return new PrismaSchema(datasource, generator);
```

Comment:
I cloned and built this branch locally, and it seems like the `index.d.ts` file needs to be updated since it currently says `createSchema` can only return a `PrivateSchema`, which is not the case anymore.

```
export declare const createSchema: ({ datasource, generator, basePath, }: CreateSchemaOptions) => PrivateSchema;
```